src/magic_folder/test/test_web.py

```diff
@@ -522,6 +526,36 @@ def test_add_folder_invalid_json(self):
             ),
         )
 
+    def test_add_folder_illegal_scan_interval(self):
+        """
+        A request for **POST /v1/magic-folder** that has a negative
+        scan_interval fails with NOT ACCEPTABLE.
+        """
+        treq = treq_for_folders(object(), FilePath(self.mktemp()), AUTH_TOKEN, {}, False)
+        self.assertThat(
+            authorized_request(
+                treq, AUTH_TOKEN, "POST", self.url, dumps({
+                    'name': 'valid',
+                    'author_name': 'author',
+                    'local_path': 'foo',
+                    'poll_interval': 60,
+                    'scan_interval': -123,
+                })
+            ),
+            succeeded(
+                matches_response(
+                    code_matcher=Equals(BAD_REQUEST),
+                    body_matcher=AfterPreprocessing(
+                        loads,
+                        MatchesDict(
+                            {
+                                "reason": StartsWith("scan_interval must be positive integer or null"),
```

Comment:
So I begin to believe `null` equals `0`, while `None` equals `not set`. Correct?

If this is so in the rest of the application, `null` as the word (as opposed to `positive integer`) is probably better English than `0` here... the digit `0` might be less confusing still